src/model/simulation.ts

```diff
@@ -310,6 +310,25 @@ function handleActions(allies: Combattant[], enemies: Combattant[], actionTypes:
     })
 }
 
+function handleDebuffRiderEffects(allies: Combattant[], stats: Map<string, EncounterStats>) {
+    allies.forEach(target => {
+        let totalDamageApplied = 0
+        target.finalState.buffs.forEach((buff, name) => {
+            if(buff.applyDamage)
+            {
+                let appliedDamage = Math.max(evaluateDiceFormula(buff.applyDamage), 0)
+                if (buff.halfOnSave)
+                    appliedDamage = appliedDamage / 2
+    
+                totalDamageApplied += appliedDamage
+            }
+        })
+        //TODO Apply damage to temp hitpoints first
+        target.finalState.currentHP = Math.min(target.finalState.currentHP, Math.max(0, target.finalState.currentHP - totalDamageApplied))
+        getStats(stats, target).damageTaken += totalDamageApplied //TODO Apply damage dealt number to owner of buff?
+    } );
+}
+
```

Comment:
The formula is wrong here - if the target takes half damage on a successful save, it should take *more* damage on average (see the comment below, on `useAtkAction` for more detail)

I think this probably needs to use the buff.magnitude field as well - this field represents the odds of the buff being present (for a rider effect, it's the hit chance of the attack, times the chance for the target to fail the save against the debuff). So if the damage debuff deals 40 damage per round, but only has 50% chance to be present, then it should only deal 20 damage, not 40.

There's probably a way to extract the damage logic from useAtkAction into a separate function, so you can re-use it here. That way we won't have to worry about temp hp, or tracking the attacker's damage dealt stat - that's already handled by the bit of code in the useAtkAction function.
This will still probably require adding a "buff.bufferId" field, so we can find the original attacker